java/all/src/main/scala/org/typelevel/otel4s/java/ContextConversions.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s.java
+
+import cats.effect.Sync
+
+import io.opentelemetry.api.trace.{Span => JSpan}
+import io.opentelemetry.context.{Context => JContext}
+import org.typelevel.otel4s.java.trace.SpanBackendImpl
+import org.typelevel.otel4s.java.trace.WrappedSpanContext
+import org.typelevel.otel4s.trace.Span
+import org.typelevel.vault.Vault
+
+private[otel4s] object ContextConversions {
```

Comment:
We're in a pickle here, because we need an isomorphism between our context and Java's to build on their propagators.  The keys aren't public.  The best we can do is translate the well-known keys.  Far as I can tell, opentelemetry-java is using it for Spans and Baggage and some weird shims.  Of course, we can't see what _extensions_ to opentelemetry-java are shoving in there.  This is damned to be an imperfect conversion.  If we make it _configurable_, we can probably live with it?

Just using Java's context exposes JContext all over the place.

This sucks, but it works, and it gets rid of a lot of Aux mess!